### PR TITLE
Job Completion Notification

### LINKED ISSUE #205: Job Completion Notification
Right now many unit tests are racy because the jobs involve shelling out to run `date` or a similar command, and there is no way for the scheduler of a job to know when that job has actually completed, so we rely on `time.Sleep()` and similar mechanisms.

This tends to create a problem for small `time.Sleep()` values or when running tests with `-count #` for testing durability.

A likely solution would be to add something like `context.Done()` to `job.Job` wherein the caller gets a channel which can be read from to block execution until the job completes.

Note: the *behavior* is not racey, but the tests are in some cases.

### PR DESCRIPTION
**Pull request by tooolbox**
Replaces https://github.com/ajvb/kala/pull/212

Fixes #205 

I think there's something odd with the CircleCI settings, but I don't have access.

### CODE REVIEW

**Comment by tooolbox** [comment]
Fantastic...

**Comment by tooolbox** [comment]
@ajvb Hey man, it *looks* like the settings for CircleCI or GitHub workflows changed somehow because now there's this additional `ci/circleci Expected — Waiting for status to be reported` that I did not see in earlier PRs.  For example, this one only has one check: https://github.com/ajvb/kala/pull/211

When I hit up Google, I see a lot of results recommending to "disable the check on the protected branch".  Unfortunately I don't have access to the branch settings in this repo.

Could you assist?  Thanks!

**Comment by tooolbox** [comment]
@ajvb So I figured out how to manually mark checks as complete to get around this particular issue, but it would still be great to fix this for the future.

Unfortunately I think this is my fault; when I clicked through to CircleCI I think it asked me if I wanted to upgrade to "Pipelines" and I hit "yes".  The resulting problem is covered in the CircleCI docs: https://circleci.com/docs/2.0/workflows-waiting-status/

Basically seems like they changed the name of the checks and now this project is looking for the wrong one.  So an admin (yourself) has to go in and fix that.

No particular rush but it would be good to resolve at some point.
